### PR TITLE
REPO-5325: TMDQ accelerated table support detection

### PR DESCRIPTION
**Pull request by n-insaidoo**
This is the community component of the TMQD accelerated table support. See the [companion PR on enterprise](https://github.com/Alfresco/alfresco-enterprise-repository/pull/478). The intent is to push as much as possible of the logic into Enterprise (being this enterprise feature) while leaving minimal possible changes in the community.

### CODE REVIEW

File: src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryBuilderPredicatePartCommand.java [hunk 1]
```diff
@@ -198,5 +201,15 @@ else if(function == LuceneFunction.UPPER)
...
+
+	public void setPropertyQName(QName propertyQName) 
+	{
+		this.propertyQName = propertyQName;
+	}
+	
+	public QName getPropertyQName() 
+	{
+		return this.propertyQName;
+	}
```
Comment (cmshields) [inline src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryBuilderPredicatePartCommand.java:213]: Use spaces instead of tabs.

**Review by cmshields — CHANGES_REQUESTED** [review]
See comments added.

File: src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngine.java [hunk 2]
```diff
@@ -205,20 +206,50 @@ public QueryEngineResults executeQuery(Query query, QueryOptions options, Functi
...
+        HashSet<String> key = new HashSet<String>();
+        key.add("");
         answer.put(key, paged);
         return new QueryEngineResults(answer);
     }
 
+    protected List<Node> selectNodes(QueryOptions options, DBQuery dbQuery)
+    {
+        List<Node> nodes = new ArrayList<Node>();
+        nodes = template.selectList(SELECT_BY_DYNAMIC_QUERY, dbQuery);
```
[the remark this answers is not in the document]
Comment (bbossola) [reply src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngine.java:249]: At the end of the day, this is a single method call "template.selectList(...)" that have just been written in a very verbose way. Following your earlier suggestion will reduce this to a one-liner, I would just do that.

File: src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryBuilderPredicatePartCommand.java (hunk 1 above)
Comment (n-insaidoo) [reply src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryBuilderPredicatePartCommand.java]: Sorted out in the last [commit](https://github.com/Alfresco/alfresco-repository/pull/1204/commits/c0752a79701821c01f98dc332bc8e673391a04aa).

**Review by cmshields — CHANGES_REQUESTED** [review]
Code looks good but this should not be merged to master until all the ACS-247 code is ready.
Could you create an alfresco-repository branch off master as feature/ACS-247_relational-query-accelerator like we have in alfresco-enterprise-repository? Once you have created the branch I'll be happy to approve this PR so that it can be merged onto the new branch.

**Review by cmshields — APPROVED** [review]
LGTM

**Pull request merged**